tools/src/publish-packages/tasks/publishCanary.ts

```diff
@@ -57,8 +57,15 @@ export const prepareCanaries = new Task<TaskArgs>(
         ? nextSdkVersion
         : resolveReleaseTypeAndVersion(parcel, options);
 
+      // Strip any pre-release tag from the baseVersion
+      // For example, convert "5.0.0-rc.0" or "5.0.0-preview.0" to "5.0.0"
+      // This is to ensure we don't stack the canary suffix on top of another
+      const cleanBaseVersion = semver.parse(baseVersion)?.version
+        ? `${semver.major(baseVersion)}.${semver.minor(baseVersion)}.${semver.patch(baseVersion)}`
+        : baseVersion;
```

Comment:
```suggestion
      const cleanBaseVersion = semver.coerce(baseVersion).version;
```

Reply:
See https://stackblitz.com/edit/node-kpj6tckw?file=index.js

Reply:
yeah i'm not sure either. i was erring on the side of being careful but maybe coerce is good enough here.